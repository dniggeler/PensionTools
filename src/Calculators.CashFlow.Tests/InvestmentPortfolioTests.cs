using System;
using System.Collections.Generic;
using PensionCoach.Tools.CommonTypes;
using PensionCoach.Tools.CommonTypes.MultiPeriod;
using PensionCoach.Tools.CommonTypes.MultiPeriod.Definitions;
using Snapshooter.Xunit;
using Xunit;

namespace Calculators.CashFlow.Tests;

[Trait("Higher Level Calculators", "Cash-Flow")]
public class InvestmentPortfolioTests
{
    [Fact(DisplayName = "Portfolio Growth")]
    public void Calculate_Growth_Of_Investment_Portfolio()
    {
        // given
        InvestmentPortfolioDefinition definition = new ()
        {
            Header = new CashFlowHeader
            {
                Id = "test",
                Name = "Test",
            },
            DateOfProcess = new DateTime(2021, 1, 1),
            NetCapitalGrowthRate = 0.02M,
            NetInterestRate = 0.01M,
            InitialInvestment = 100_000,
            RecurringInvestment = new RecurringInvestment
            {
                Amount = 6723,
                Frequency = FrequencyType.Yearly,
            },
            InvestmentPeriod = new InvestmentPeriod
            {
                Year = 2021,
                NumberOfPeriods = 10
            },
        };

        // when
        IEnumerable<ICashFlowDefinition> result = definition.CreateGenericDefinition();

        // then
        Snapshot.Match(result);
    }
}
