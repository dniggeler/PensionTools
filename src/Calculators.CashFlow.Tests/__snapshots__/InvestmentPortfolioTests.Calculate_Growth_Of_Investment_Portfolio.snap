[
  {
    "Header": {
      "Id": "CapitalGrowth",
      "Name": "Portfolio's Capital Growth"
    },
    "DateOfProcess": "2021-01-01T00:00:00",
    "InitialAmount": 100000.0,
    "RecurringInvestment": {
      "Amount": 6723.0,
      "Frequency": "Yearly"
    },
    "NetGrowthRate": 0.02,
    "InvestmentPeriod": {
      "Year": 2021,
      "NumberOfPeriods": 10
    },
    "Flow": {
      "Source": "Exogenous",
      "Target": "Investment"
    },
    "IsTaxable": true,
    "TaxType": "Wealth"
  },
  {
    "Header": {
      "Id": "InterestPayment",
      "Name": "Portfolio's Interest"
    },
    "DateOfProcess": "2021-01-01T00:00:00",
    "InitialAmount": 100000.0,
    "RecurringInvestment": {
      "Amount": 6723.0,
      "Frequency": "Yearly"
    },
    "NetGrowthRate": 0.01,
    "InvestmentPeriod": {
      "Year": 2021,
      "NumberOfPeriods": 10
    },
    "Flow": {
      "Source": "Exogenous",
      "Target": "Investment"
    },
    "IsTaxable": true,
    "TaxType": "Income"
  }
]
